kit/ChildSession.cpp

```diff
@@ -2360,6 +2361,9 @@ bool ChildSession::saveAs(const char* /*buffer*/, int /*length*/, const StringVe
         const std::string tmpDir = FileUtil::createRandomDir(jailDoc);
         const Poco::Path filenameParam(pathSegments[pathSegments.size() - 1]);
         url = std::string("file://") + jailDoc + tmpDir + '/' + filenameParam.getFileName();
+        // url becomes decoded at this stage
+        // on saveAs we should send encoded!
+        encodeURL = true;
```

Comment:
Wouldn't it be simpler to decode here, and always decode below?

Reply:
Ah, when we do that it has become double decoded. say the original file name is a%20b and it goes to the this place as a%2520b and we decode here it as you suggest and it becomes a%20b after that  we decode for the second time as explained for the wopi case it has become a b then we encode always as you suggest, it would become a%20b and finally after saveAs it will be a b but the original filename was a%20b. But this patch we only encode the second time if the file is decoded for the wopi case. For the other cases, it is already encoded and we dont do anything.